cache/blobs_linux.go

```diff
@@ -40,9 +41,12 @@ func (sr *immutableRef) tryComputeOverlayBlob(ctx context.Context, lower, upper
 	if err != nil {
 		return emptyDesc, false, errors.Wrap(err, "failed to open writer")
 	}
+	var openErr error // can be set before closing writer. This can be used to make sure the writer is closed.
```

Comment:
cleaner way to do this would be to set `cw = nil` after it has been closed and add a nil check if the defer. I'm not sure I understand how this is related to the issue though.